src/openeo_test_suite/tests/processes/processing/test_example.py

```diff
@@ -154,110 +201,117 @@ def prepare_argument(arg, process_id, name, connection, file):
         for i in range(len(example)):
             if i >= len(result):
-                (example[i], _) = prepare_results(connection, file, example[i])
+                (example[i], _) = _prepare_results(
+                    connection=connection, file=file, example=example[i]
+                )
             else:
-                (example[i], result[i]) = prepare_results(
-                    connection, file, example[i], result[i]
+                (example[i], result[i]) = _prepare_results(
+                    connection=connection,
+                    file=file,
+                    example=example[i],
+                    result=result[i],
                 )
 
     return (example, result)
 
 
-def load_ref(ref, file):
+def _load_ref(ref: str, file: Path):
     try:
-        path = posixpath.join(file.parent, ref)
+        path = file.parent / ref
         if ref.endswith(".json") or ref.endswith(".json5") or ref.endswith(".geojson"):
             with open(path) as f:
                 return json5.load(f)
         elif ref.endswith(".txt") or ref.endswith(".wkt2"):
             with open(path) as f:
                 return f.read()
         else:
-            raise Exception(
+            raise NotImplementedError(
                 "External references to files with the given extension not implemented yet."
             )
     except Exception as e:
-        raise Exception("Failed to load external reference {}: {}".format(ref, e))
+        # TODO: is this try-except actually useful?
+        raise Exception(f"Failed to load external reference {ref}") from e
 
 
 def check_non_json_values(value):
+    # TODO: shouldn't this check be an aspect of Http(ProcessTestRunner)?
     if isinstance(value, float):
         if math.isnan(value):
-            raise Exception("HTTP JSON APIs don't support NaN values")
+            raise ValueError("HTTP JSON APIs don't support NaN values")
         elif math.isinf(value):
-            raise Exception("HTTP JSON APIs don't support infinity values")
+            raise ValueError("HTTP JSON APIs don't support infinity values")
     elif isinstance(value, dict):
-        for key in value:
-            check_non_json_values(value[key])
+        for v in value.values():
+            check_non_json_values(v)
     elif isinstance(value, list):
         for item in value:
             check_non_json_values(item)
 
 
 def check_exception(example, result):
-    assert isinstance(result, Exception), "Excpected an exception, but got {}".format(
-        result
-    )
+    assert isinstance(result, Exception), f"Expected an exception, but got {result}"
     if isinstance(example["throws"], str):
         if result.__class__.__name__ != example["throws"]:
-            warnings.warn(
-                "Expected exception {} but got {}".format(
-                    example["throws"], result.__class__.__name__
-                )
+            # TODO: better way to report this warning?
+            _log.warning(
```

Comment:
I don't get this warning anylonger in the logs, I tried before this PR and after and these ones are missing. Any clue why?

Reply:
logs of successful tests are hidden by default. 
logs of failing tests should be visible

you can disable capturing of all logs (and print statements) with options `--capture=no --log-cli-level=INFO`

Reply:
But that shouldn't have changed. The number of successful/failing tests is the same, just the log is missing.

Reply:
do you have an example use case (process that fails on some implementation) for this, so I can play with this too?

Reply:
Run `pytest src/openeo_test_suite/tests/processes/processing/test_example.py --runner=dask` both on this branch and on master. 

You can then see differences:
![grafik](https://github.com/Open-EO/openeo-test-suite/assets/8262166/d5f5c9a8-5788-4e7d-8cd3-12ceeeedeb79)